spinnman/utilities/io/memory_io.py

```diff
@@ -197,7 +208,8 @@ class MemoryIO(AbstractIO):
     # ensure they are written before a read occurs
     __write_cache__ = dict()
 
-    def __init__(self, transceiver, x, y, start_address, end_address):
+    def __init__(self, transceiver, x, y, memory_write_function, start_address,
+                 end_address):
```

Comment:
It's a shame that we've got an incompatible API change here.

Reply:
worth thinking about, currently safe though, as all uses have been covered...... what would be declaire as default? data in or sdp?

Reply:
`transceiver.write_memory` is the backward-compatible approach. (That forces putting the parameter at the end, but that's OK.)